engineapi/snapshot.go

```diff
@@ -78,15 +79,18 @@ func (e *Engine) SnapshotPurge() error {
 	return nil
 }
 
-func (e *Engine) SnapshotBackup(snapName, backupTarget string, labels map[string]string) error {
+func (e *Engine) SnapshotBackup(snapName, backupTarget string, labels map[string]string, secret map[string]string) error {
```

Comment:
`credential`, not `secret`.

`Secret` is where Kubernetes keeps encrypted data.

`Credential` is the thing we need to access something, S3 in this case.